index.js

```diff
@@ -99,6 +99,13 @@
          * linked to this component's state.
          */
         componentDidMount: function() {
+          // If we are in an environment without a DOM such 
+          // as shallow rendering or snapshots then we exit 
+          // early to prevent any unhandled errors being thrown.
+          if (!ReactDOM.findDOMNode){
```

Comment:
I'm not entirely sure if it's the most authoritative route. I did find https://www.npmjs.com/package/exenv which is a port of what React uses to determine the environment that could be used to do the detection instead of this.

I'm looking through ReactDOM and findDOMNode but it's not yet immediately clear why it's not being required in at all in the non-DOM case. If I look at ReactDOM in this state the only method available off of it is `render`, neither `findDOMNode`, nor `version` or any of the others are available.

Let me switch this out with the environment check and see if it's still detecting it properly.